extensions/sftp/client/SFTPClient.h

```diff
@@ -49,6 +50,12 @@ enum class SFTPError : uint8_t {
   SFTP_ERROR_UNEXPECTED
 };
 
+struct SFTPException : Exception {
+  explicit SFTPException(const SFTPError err)
+      :Exception{ExceptionType::FILE_OPERATION_EXCEPTION, fmt::format("SFTP Error {0}", static_cast<int>(err))}
```

Comment:
I am not saying that we should change it in this PR (or in any PR for that matter), but turning `SFTPError` into a `SMART_ENUM` could benefit us, (e.g. here we could then write `err.toString()`)

Reply:
this is just something to think about, not a change request

Reply:
I thought about it but then went for the easier solution. I might change it after I figure out the reason for the test failures.

Reply:
I have some negative feedback regarding the conversion to SMART_ENUM: the enum values are implicitly convertible to integer which makes it less safe than `enum class`. This resulted in compiling but invalid code when the wrong overload of `LastSFTPError::operator=` was called and I had to introduce differently named functions.

I'm leaning towards keeping SMART_ENUM in this case but I'll be more careful about introducing it in new places, weighing the convenient string conversion against the safety tradeoff in each case.

Reply:
yes, the implicit convertibility is indeed troublesome (and the extra scope specifier we save when accessing an instance might not be worth it), although in this specific case I would have gone with changing the parameter type from `const SFTPError&` to `SFTPError::Type`